utils/measure/measure.py

```diff
@@ -274,6 +279,38 @@ def start(self):
 
         if bool(answers.get("gzip", True)):
             self.gzip_csv(csv_file_path)
+    def nudge_and_remeasure(self, color_mode, variation):
+        for nudge_count in range(MAX_NUDGES):
+            try:
+                # Likely not significant enough change for PM to detect. Try nudging it
+                _LOGGER.warning("Measurement is stuck, Nudging")
+                # If brightness is low, set brightness high. Else, turn light off
+                self.light_controller.change_light_state(MODE_BRIGHTNESS, on=(variation.bri < 128), bri=255)
+                time.sleep(PULSE_TIME_NUDGE)
+                variation_start_time = time.time()
+                self.light_controller.change_light_state(
+                    color_mode, on=True, **asdict(variation)
+                )
+                # Wait a longer amount of time for the PM to settle
+                time.sleep(SLEEP_TIME_NUDGE)
+                power = self.take_power_measurement(variation_start_time)
+            except OutdatedMeasurementError as error:
+                continue
+            except ZeroReadingError as error:
+                self.num_0_readings += 1
+                _LOGGER.warning(f"Discarding measurement: {error}")
+                if self.num_0_readings > MAX_ALLOWED_0_READINGS:
+                    _LOGGER.error("Aborting measurement session. Received too many 0 readings")
+                    return
+                continue
+            except PowerMeterError as error:
+                _LOGGER.error(f"Aborting: {error}")
+                return
+            except error:
+                raise error
+            else:
```

Comment:
You can remove this else, and move return power after line 296 inside the `try` statement.